searx/webapp.py

```diff
@@ -224,6 +227,21 @@
 _flask_babel_get_translations = flask_babel.get_translations
 
 
+class ExtendedRequest(flask.Request):
+    """This class is never initialized and only used for type checking."""
+
+    preferences: Preferences
+    errors: List[str]
+    user_plugins: List[Plugin]
+    form: Dict[str, str]
+    start_time: float
+    render_time: float
+    timings: List[Timing]
+
```

Comment:
On the long term we should modularize the webapp.py .. can we place this type definition in a different module or create a new module if nothing fits?

Reply:
Right, I'd prefer us to move it when the time comes instead of now.